### PR TITLE
Add TransactionWitness.redeemP2WSH()

### PR DESCRIPTION
**Pull request by oscarguindzberg**

### CODE REVIEW

**Comment by schildbach** [comment]
First of all, I think this really needs a unit test.

When it comes to P2WSH, it is my understanding that it is a successor to P2SH, right? This is why I'm always looking for how P2SH did things API-wise. In this case, I could not find an equivalent. Do you know any?

File: core/src/main/java/org/bitcoinj/core/TransactionWitness.java [hunk 1]
```diff
@@ -41,6 +41,20 @@ public static TransactionWitness redeemP2WPKH(@Nullable TransactionSignature sig
         return witness;
     }
 
+    /**
+     * Creates the stack pushes necessary to redeem a P2WSH output.
+     */
+    public static TransactionWitness redeemP2WSH(byte[] witnessScript, TransactionSignature... signatures) {
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/TransactionWitness.java:47]: Would it maybe make sense to express the witnessScript as type Script?

Comment (oscarguindzberg) [reply core/src/main/java/org/bitcoinj/core/TransactionWitness.java:47]: fixed

**Comment by oscarguindzberg** [comment]
Added a unit test.

In the future it would be good to write tests for the examples described in the BIP https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#Native_P2WSH .

The P2SH way of doing this is ScriptBuilder.createP2SHMultiSigInputScript()

**Comment by schildbach** [comment]
Merged.

**Comment by schildbach** [comment]
> In the future it would be good to write tests for the examples described in the BIP https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#Native_P2WSH .

Yes, I tried to follow those example with P2WPKH too.

**Pull request closed without merging**